### PR TITLE
[KOGITO-2932] Install operators one after the other

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2932

Due to https://github.com/operator-framework/operator-lifecycle-manager/issues/1704

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by radtriste** [comment]
Jenkins test this

**Comment by radtriste** [comment]
also tested the full testing, no more `Timeout waiting for operator`

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
@Sgitario hard to tell because some other tests failed because of other reasons ... but timeout for Strimzi operator is 30 min (load_factor is 3 by default) so each time there is a problem on operator, then it waits 30 min before failing.
I would expect the time installing operators taking less than 30 min overall ;)

But I can make it configurable if you prefer, with sequential by default for now. wdyt ?

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins test this

**Comment by Sgitario** [comment]
> @Sgitario hard to tell because some other tests failed because of other reasons ... but timeout for Strimzi operator is 30 min (load_factor is 3 by default) so each time there is a problem on operator, then it waits 30 min before failing.
> I would expect the time installing operators taking less than 30 min overall ;)
> 
> But I can make it configurable if you prefer, with sequential by default for now. wdyt ?

If it would only work sequentially, then it's fine to not being configurable. But if the all at once installation works for some scenario, then I would expect to be configurable.

**Comment by radtriste** [comment]
well this an olm issue and at schrödinger installation: it may or may not work ...
it is a problem if number of dependent operators >= 2. I could change to check that.

File: test/steps/operator.go [hunk 1]
```diff
@@ -61,32 +61,30 @@ func (data *Data) kogitoOperatorIsDeployedWithDependencies(dependencies string)
...
-			installedOperators = append(installedOperators, dependentOperator)
+			if err := framework.WaitForKogitoOperatorDependencyRunning(data.Namespace, dependentOperator, operatorSource); err != nil {
+				return err
+			}
 		}
 	}
 
-	// Wait for kogito operator to be running
+	// Wait for Kogito operator running
+	// This is put at the end because there should not be any racing condition as install is done via yaml and not OLM
```
Comment (radtriste) [inline test/steps/operator.go:84]: @Sgitario I let this one at the end as there should not be any conflict. no need to wait first for Kogito Operator running

**Comment by radtriste** [comment]
Jenkins test this

**Pull request merged**